### DISCUSSION TITLE
Feature Request: Commenting System similar to Notion/Google Docs

### DISCUSSION THREAD

**Discussion by yathindrak** [steven-tey/novel#289] [Ideas]
A commenting system for Novel would be great IMO ✨. 

- It allows users to highlight text and shoot in some hot takes🔥 or constructive feedback, Notion or Google Docs style.  directly inline.
- Then multiple users comment on the same portion of the text 🗣️, the comments can be threaded.
- Once feedback has been addressed, users can have the ability to mark a comment as resolved ✅. Resolved comments can then either disappear or be accessible in a resolved comments section/archive.

**Comment by fullstackjedi** [comment]
I implemented something similar in my project using this as a guide - https://github.com/sereneinserenade/tiptap-comment-extension.

Where it gets tricky if you'd like to sync these comments to a DB, remove a comment from the DB when they are removed from the editor, and if people can create comments that aren't linked to a piece of text in the editor etc.

**Comment by sereneinserenade** [comment]
I puhlished it to npm, so it can be used directly https://github.com/sereneinserenade/tiptap-comment-extension

```bash
npm i @sereneinserenade/tiptap-comment-extension
```
